File: algorithms/1_easy/S/01629_slowest_key/solution.c
char slowestKey(const int *restrict release_times, const int release_times_len, const char *restrict keys_pressed) {
    int prev = 0;
    int duration[256] = { 0 };
    for (int i = 0; i < release_times_len; i += 1) {
        duration[(unsigned char)keys_pressed[i]] =  MAX(duration[(unsigned char)keys_pressed[i]], release_times[i] - prev);
        prev = release_times[i];
    }

    int longest = 0;
    char longest_char = 0;
    for (int i = 0; i < 255; i += 1) {
        if (duration[i] >= longest) {
            longest = duration[i];
            longest_char = (char)i;
        }
    }

    return longest_char;
}

File: algorithms/1_easy/S/01629_slowest_key/solution.rs
impl Solution {
    pub fn slowest_key(release_times: Vec<i32>, keys_pressed: String) -> char {
        let mut prev: i32 = 0;
        let mut duration = [0_i32; 256];
        for (i, &b) in keys_pressed.as_bytes().into_iter().enumerate() {
            duration[b as usize] =  duration[b as usize].max(release_times[i] - prev);
            prev = release_times[i];
        }

        let mut longest: i32 = 0;
        let mut longest_byte: u8 = 0;
        for b in 0..255 {
            if duration[b] >= longest {
                longest = duration[b];
                longest_byte = b as u8;
            }
        }

        longest_byte as char
    }
}
